Intersect.Client.Core/Interface/Game/Bag/BagItem.cs

```diff
@@ -141,47 +122,69 @@ private void _iconImage_HoverEnter(Base? sender, EventArgs? arguments)
         );
     }
 
-    private void _iconImage_HoverLeave(Base sender, EventArgs arguments)
+    private void IconImage_HoverLeave(Base sender, EventArgs arguments)
     {
-        _mouseOver = false;
-        _mouseX = -1;
-        _mouseY = -1;
+        _descWindow?.Dispose();
+        _descWindow = default;
+    }
 
-        if (_descWindow != default)
+    private void IconImage_Clicked(Base sender, MouseButtonState arguments)
+    {
+        if (arguments.MouseButton is MouseButton.Right)
         {
-            _descWindow.Dispose();
-            _descWindow = default;
+            if (ClientConfiguration.Instance.EnableContextMenus)
+            {
+                OpenContextMenu();
+            }
+            else
+            {
+                IconImage_DoubleClicked(sender, arguments);
+            }
         }
     }
 
-    private void _iconImage_Clicked(Base sender, MouseButtonState arguments)
+    private void IconImage_DoubleClicked(Base sender, MouseButtonState arguments)
     {
-        // ReSharper disable once SwitchStatementMissingSomeEnumCasesNoDefault
-        switch (arguments.MouseButton)
+        if (Globals.InBag)
         {
-            case MouseButton.Right:
-                if (ClientConfiguration.Instance.EnableContextMenus)
-                {
-                    OpenContextMenu();
-                }
-                else
-                {
-                    _iconImage_DoubleClicked(sender, arguments);
-                }
-                break;
-
-            case MouseButton.Left:
-                _clickTime = Timing.Global.MillisecondsUtc + 500;
-                break;
+            Globals.Me?.TryRetrieveItemFromBag(SlotIndex, -1);
         }
     }
 
-    private void _iconImage_DoubleClicked(Base sender, MouseButtonState arguments)
+    #endregion
+
+    #region Drag and Drop
+
+    public override bool DragAndDrop_HandleDrop(Package package, int x, int y)
     {
-        if (Globals.InBag)
+        var targetNode = Interface.FindComponentUnderCursor(NodeFilter.None);
```

Comment:
You don't need `NodeFilter.None`, the parameter is optional and `None` is the default.